packages/flutter_driver/test/src/real_tests/timeline_test.dart

```diff
@@ -56,5 +56,53 @@ void main() {
       expect(e2.threadTimestampMicros, isNull);
       expect(e2.arguments, isNull);
     });
+
+    test('sorts JSON', () {
+      final Timeline timeline = Timeline.fromJson(<String, dynamic>{
+        'traceEvents': <Map<String, dynamic>>[
+          <String, dynamic>{
+            'name': 'test event 1',
+            'ts': 456,
+          },
+          <String, dynamic>{
+            'name': 'test event 2',
+            'ts': 457,
+          },
+        ],
+      });
+
+      expect(timeline.events, hasLength(2));
+      expect(timeline.events[0].timestampMicros, equals(456));
+      expect(timeline.events[1].timestampMicros, equals(457));
```

Comment:
It couldn't hurt to test the names as well as the timestamps. See comment on next test...